src/bun.js/bindings/bindings.cpp

```diff
@@ -2497,9 +2497,13 @@ JSC__JSValue JSC__JSValue__keys(JSC__JSGlobalObject* globalObject, JSC__JSValue
 bool JSC__JSValue__hasOwnProperty(JSC__JSValue jsValue, JSC__JSGlobalObject* globalObject, ZigString key)
 {
     JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
 
     JSC::JSValue value = JSC::JSValue::decode(jsValue);
-    return value.toObject(globalObject)->hasOwnProperty(globalObject, JSC::PropertyName(JSC::Identifier::fromString(vm, Zig::toString(key))));
+    JSObject* obj = value.toObject(globalObject);
+    RETURN_IF_EXCEPTION(scope, false);
+
+    RELEASE_AND_RETURN(scope, obj->hasOwnProperty(globalObject, JSC::PropertyName(JSC::Identifier::fromString(vm, Zig::toString(key)))));
```

Comment:
->hasOwnProperty can throw when its a proxy and I think its somewhat ambiguous whether it would throw here

Reply:
yeah, I think in this case it shouldn't catch the exception because `toContainKey(s)` uses `hasOwnProperty` directly, but other internal uses of this function will probably want to catch it

Reply:
If it was a proxy it would cause `Object.hasOwn` to throw and `.hasOwnProperty('x')` to throw. I think it's fine to blow up here too.. we can do what WebKit does and 

https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/runtime/ObjectConstructor.cpp#L1368-L1377
then goes to:
https://github.com/WebKit/WebKit/blob/e954cf3227f1d0672e1d61645e37400059d29e58/Source/JavaScriptCore/runtime/ObjectPrototype.cpp#L103-L121

Which does capture the ` thisObject->hasOwnProperty` in a throw scope. It also leverages a cache which is neeat.

Reply:
The JS implementation does an odd check,  if the object has a truthy `.hasOwnProperty` (so not a null prototype) and then calls`Object.prototype.hasOwnProperty` so works with a `null` prototype. All instead using `Object.hasOwn`. We could PR that. https://github.com/jest-community/jest-extended/blob/711fdcc54d68c2b2c1992c7cfbdf0d0bd6be0f4d/src/matchers/toContainKeys.js#L1-L6

Reply:
The JS implementation allows `falsey` to pass through meaning they don't allow things like `''` or `0` to go through. I'm guessing their intention is to really only support objects but because JS is wiggly other stuff like strings is doable.

Reply:
> Which does capture the ` thisObject->hasOwnProperty` in a throw scope. It also leverages a cache which is neeat.

oh nice, yeah we should use `objectPrototypeHasOwnProperty()` for the cache

Reply:
> The JS implementation does an odd check, if the object has a truthy `.hasOwnProperty` (so not a null prototype) and then calls`Object.prototype.hasOwnProperty` so works with a `null` prototype. All instead using `Object.hasOwn`. We could PR that. https://github.com/jest-community/jest-extended/blob/711fdcc54d68c2b2c1992c7cfbdf0d0bd6be0f4d/src/matchers/toContainKeys.js#L1-L6

yep, will update to check for this before `hasOwnProperty` else it will throw when it shouldn't